Description: Fix background color drawing
Author: Jens Georg <mail@jensge.org>
Origin: https://git.gnome.org/browse/shotwell/commit/?id=7361b3f
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=766864
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=823835
Forwarded: not-needed
Last-Update: 2015-06-05
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: trunk/src/CheckerboardLayout.vala
===================================================================
--- trunk.orig/src/CheckerboardLayout.vala
+++ trunk/src/CheckerboardLayout.vala
@@ -1812,6 +1812,10 @@ public class CheckerboardLayout : Gtk.Dr
         // we want switched_to() to be the final call in the process (indicating that the page is
         // now in place and should do its thing to update itself), have to be be prepared for
         // GTK/GDK calls between the widgets being actually present on the screen and "switched to"
+
+        Gtk.Allocation allocation;
+        get_allocation(out allocation);
+        get_style_context().render_background (ctx, 0, 0, allocation.width, allocation.height);
         
         // watch for message mode
         if (message == null) {
@@ -1833,7 +1837,6 @@ public class CheckerboardLayout : Gtk.Dr
             int text_width, text_height;
             pango_layout.get_pixel_size(out text_width, out text_height);
             
-            Gtk.Allocation allocation;
             get_allocation(out allocation);
             
             int x = allocation.width - text_width;
